pkg/cmd/cli/cmd/deploy.go

```diff
@@ -352,6 +361,9 @@ func (o DeployOptions) cancel(config *deployapi.DeploymentConfig, out io.Writer)
 
 // reenableTriggers enables all image triggers and then persists config.
 func (o DeployOptions) reenableTriggers(config *deployapi.DeploymentConfig, out io.Writer) error {
+	if config.Spec.Paused {
+		return fmt.Errorf("cannot enable triggers for a paused deployment config")
```

Comment:
Why are we not modeling paused as a new field that overrides trigger state?

On Mon, Jun 6, 2016 at 9:53 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/deploy.go
> <https://github.com/openshift/origin/pull/9086#discussion_r65893001>:
>
> > @@ -352,6 +361,9 @@ func (o DeployOptions) cancel(config *deployapi.DeploymentConfig, out io.Writer)
> >
> >  // reenableTriggers enables all image triggers and then persists config.
> >  func (o DeployOptions) reenableTriggers(config *deployapi.DeploymentConfig, out io.Writer) error {
> > +	if config.Spec.Paused {
> > +		return fmt.Errorf("cannot enable triggers for a paused deployment config")
>
> We want paused to have triggers set to automatic:false. I think we need
> to enforce it on the server too. When a deployment config is paused then we
> shouldnt' allow updates on automatic: false to automatic: true or the
> addition of a config change trigger.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9086/files/3d2699cf1c43a644b1af175d23b5291c6956a708#r65893001>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p_VdGeCmPZZVO09qsqDM5K1pNQYZks5qJCZ0gaJpZM4Ip6MH>
> .
>